### PR TITLE
[WIP] NC-569

### PR DESCRIPTION
**Pull request by Joeper214**
Ref: https://skyarch.backlog.jp/view/NC-569

### Goal of this PR.

- [X] Validation for EC2 if instance keypair is not matched with the infrastructure keypair.
- [x] Validation for EC2 if instance is terminated.

### CODE REVIEW

File: app/controllers/resources_controller.rb [hunk 1]
```diff
@@ -49,6 +49,16 @@ def create
...
     end
 
+    unless infra.instance(physical_id).describe_keypair == infra.keypairname
+      # TODO: I18n
+      render text: "#{physical_id}: Keypair does not matched to this Infrastructure!", status: 400; return
+    end
+
+    unless infra.instance(physical_id).status != :terminated
+      # TODO: I18n
+      render text: "Cannot add #{physical_id}. if status is terminated.", status: 400; return
```
Comment (skyarch-networks) [inline app/controllers/resources_controller.rb]: __[brakeman]__
* Unescaped parameter value rendered inline [:ref](http://brakemanscanner.org/docs/warning_types/cross_site_scripting/)

File: app/controllers/resources_controller.rb [hunk 2]
```diff
@@ -49,6 +49,16 @@ def create
       render text: "Cannot find #{physical_id}", status: 400; return
     end
 
+    unless infra.instance(physical_id).describe_keypair == infra.keypairname
+      # TODO: I18n
+      render text: "#{physical_id}: Keypair does not matched to this Infrastructure!", status: 400; return
```
Comment (skyarch-networks) [inline app/controllers/resources_controller.rb]: __[brakeman]__
* Unescaped parameter value rendered inline [:ref](http://brakemanscanner.org/docs/warning_types/cross_site_scripting/)

**Comment by Joeper214** [comment]
Sorry test is failed. I'll try to fix it first.

**Comment by Joeper214** [comment]
Test is still broken.

**Comment by Joeper214** [comment]
Please help fix test. @pocke

**Comment by pocke** [comment]
@Joeper214 Is problem test only ?

**Comment by Joeper214** [comment]
Yes.

**Comment by pocke** [comment]
test is fixed

**Comment by Joeper214** [comment]
thanks!. Please review and Merge PR. @pocke

**Comment by pocke** [comment]
LGTM :100:

**Comment by Joeper214** [comment]
Thanks!

**Pull request merged**